### PR TITLE
[shuffler] retry transient errors

### PR DESCRIPTION
**Pull request by danking**
I also added SocketException and SSLException as transient errors

### CODE REVIEW

**Comment by danking** [comment]
Looks better with whitespace changes hidden: https://github.com/hail-is/hail/pull/9404/files?diff=unified&w=1

File: router/router.nginx.conf.in [hunk 1]
```diff
@@ -75,8 +75,8 @@ server {
         include /etc/nginx/proxy.conf;
     }
 
-    listen 80;
-    listen [::]:80;
+    listen 443;
```
Comment (jigold) [inline router/router.nginx.conf.in:78]: Can you put this in a different PR?

**Review by jigold — CHANGES_REQUESTED** [review]

File: hail/src/test/scala/is/hail/services/shuffler/ShuffleSuite.scala [hunk 2]
```diff
@@ -209,7 +212,7 @@ class ShuffleSuite extends HailSuite {
         .asInstanceOf[PArray]
       val elementPType = rowArrayPType.elementType
         .asInstanceOf[PBaseStruct]
-      val pairOff = f(0, ctx.r)(ctx.r)
+      val pairOff == retryTransientErrors { f(0, ctx.r)(ctx.r) }
```
Comment (jigold) [inline hail/src/test/scala/is/hail/services/shuffler/ShuffleSuite.scala:215]: I think there's an extra equal sign here.

**Review by jigold — DISMISSED** [review]

File: router/router.nginx.conf.in (hunk 1 above)
Comment (danking) [reply router/router.nginx.conf.in:78]: huh, sorry, not sure how that got in here.

File: router/router.nginx.conf.in [hunk 3]
```diff
@@ -75,8 +75,8 @@ server {
         include /etc/nginx/proxy.conf;
     }
 
-    listen 443;
-    listen [::]:443;
+    listen 80;
```
Comment (jigold) [inline router/router.nginx.conf.in]: Why are there changes to this in this PR?

**Review by jigold — CHANGES_REQUESTED** [review]

**Review by jigold — APPROVED** [review]
